src/root/rmem.d

```diff
@@ -209,3 +209,18 @@ else
         }
     }
 }
+
+extern (D) static char[] xarraydup(const(char)[] s) nothrow
+{
+    if (s)
```

Comment:
sigh again with the arrow

```d
if (!s)
    return null; // and we're done
... bulk comes here ...
```

With the current style the decision to return null for null input is completely separated from its hypothesis.